### ISSUE TITLE
Create new release with django v4 support

### ISSUE THREAD

**Issue by JanMalte** [django-haystack/django-haystack#1863]
Please publish a new release with django v4 support, which is already merged into the `master` branch.

See: #1826

**Comment by acdha** [comment]
New GitHub release here: https://github.com/django-haystack/django-haystack/releases/tag/v3.3b1

I might need to update this if I missed part of the PyPI action setup. I'm trying to bring this into alignment with pysolr.

**Comment by dustinblanchard** [comment]
@cclauss what's the timeframe for incorporating this in a PyPl release?

**Comment by cclauss** [comment]
@dustinblanchard There are three pinned issues at https://github.com/django-haystack/django-haystack/issues
Once the first two are fixed then I think we should be able to progress on this one.

**Comment by darkpixel** [comment]
Django 5 has been out for about 4 months now and Elasticsearch 8.x has been out for over 13 months.
I've tried installing Haystack from master and it's pretty bad.

There's no backend support for Elasticsearch 8.

The Elasticsearch 7 backend might be working, but I get non-stop SSL/TLS errors from urllib3 with self-signed certs.  The fix that worked with previous versions mentioned in #1678 no longer works and appears to be due to the python elasticsearch library changing...but there doesn't appear to be a way to influence certificate handling in django-haystack backends.  I basically have to pin elasticsearch to 7.17.9 and urllib3 to 1.26.18.  If I install the latest versions, nothing works.

**Comment by black3r** [comment]
Hi, I'd like to help get this done since this is the last package blocking our upgrade to Django 5.0 and Python 3.12... and I'm wondering what's the best place to start with and how to contribute. 

I figured #1917 or #1921 would be the places to start, but I see the unresolved merge conflicts are mostly about which versions of Python & Django to support going forward where I as an outsider don't feel like a worthy judge to pick what's best... 

#1921 sets the CI to test for Django 4.1 and python 3.11 only. While master currently tests for Django 3.2, 4.2 and 5.0, but only on Python 3.8 and 3.9. We're currently using django-haystack 3.2.1 with whoosh and Django 4.2 and Python 3.11 without issues and plan to upgrade to Django 5.0 and Python 3.12 and neither of those combinations are currently tested in either master or #1921. 

I'm willing to look at the tests and failing CI, but I need to know which versions of Django and Python I should focus on and I feel like that's a decision that should be made by the project's maintainer, but I'm not even sure who the current maintainer is. 

(pinging @acdha @cclauss as you seem to be the most active people concerning these issues)

**Comment by cclauss** [comment]
It would be great to have your help on this.

The three pinned issues at the top of our issues page are meant to be solved from left to right.
* https://github.com/django-haystack/django-haystack/issues
    * #1917 
    * #1924
    * #1863

**Comment by black3r** [comment]
Seems like tests are currently passing for python 3.12 and django 5.0. Is there anything other blocking a new release supporting these targets that I could help with?

**Comment by acdha** [comment]
Ah, cute: adding a protected tag rule does _not_ cause the GitHub Actions environments to process it with the default “protected branches and tags” setting but setting the same pattern there did produce a completely untested result: 

https://pypi.org/project/django-haystack/3.3b2/

**Comment by cclauss** [comment]
What is the testing process beyond the GitHub Actions?

That is a beta release so we could encourage interested parties to test it too.

**Comment by cclauss** [comment]
The docs need to be updated.

**Comment by acdha** [comment]
> What is the testing process beyond the GitHub Actions?
> 
> That is a beta release so we could encourage interested parties to test it too.

Yes - I think at this point we probably to do some basic testing, ask people who've been on some of those issues to confirm success for them, and start working on the docs updates and release notes. I did the lazy auto-generated version for v3.3b2 but it could be cleaned up more.

I think the good news is that since the publication process is now very simple we'll be able to do releases easily in the future.

**Comment by claudep** [comment]
I tested 3.3b2 in a project with the xapian backend (https://pypi.org/project/xapian-haystack/) and all tests pass with Django 5. Looks good!

**Comment by cclauss** [comment]
@black3r Can you please do the same with Whoosh?

**Comment by acdha** [comment]
I'm thinking that in the absence of issues reported and @claudep's positive report, we can cut the release.

**Comment by acdha** [comment]
https://github.com/django-haystack/django-haystack/releases/tag/v3.3.0
